reactioncommerce/reaction
# #5097: 📬 Order confirmation email can be overriden by a Plugin

Resolves #5097  
Impact: **critical**  
Type: **feature**

## Issue
Plugins can trigger emails, like the Order Confirmation email, to be sent after a certain `appEvent`, but they currently cannot cancel the API's default email from being sent. As a result, in the current app, if a custom plugin sends a custom Order Confirmation email, the customer will get 2 emails: Reaction's default, and the custom one. A plugin developer should be able to _override_ the default Order Confirmation email method, rather than add its own.

## Solution
- Refactor the current `core/orders/server/util/sendOrderEmail.js` method to use a new `core/orders/server/no-meteor` mutation method instead.
- Plugins can now override this `sendOrderEmail` mutation.

## Breaking changes
None

## Testing
0. Make sure your stock StarterKit supports email (SMTP) and checkout (payment). (Ask a coworker for SMTP config instructions).
1. In stock Reaction and stock Starterkit, make sure Reaction Order confirmation emails still work
2. Make sure a plugin can override this mail function successfully (to be tested later more fully in the plugin itself.)
